### PR TITLE
Improve Syncing Obsidian Vault

### PR DESCRIPTION
**Pull request by debanjum**
### Improve
- Only send files modified since their last sync for indexing on server from the Obsidian client

### Fix
- Invalidate static asset browser cache in Web client when Khoj version changes

### CODE REVIEW

File: src/interface/obsidian/src/utils.ts [hunk 1]
```diff
@@ -28,17 +28,42 @@ function fileExtensionToMimeType (extension: string): string {
...
+        case 'jpg':
+        case 'jpeg':
+            return 'image/jpeg';
+        case 'md':
+        case 'markdown':
+            return 'text/markdown';
+        case 'org':
+            return 'text/org';
+        default:
+            return 'text/plain';
```
Comment (sabaimran) [inline src/interface/obsidian/src/utils.ts:46]: hmm, is this a fair assumption? there would be a bunch of file types we can't process, and we wouldn't necessarily want them to slip through if they're interpreted as `text/plain`. like docx, xml, odt, ppt, etc.

File: src/khoj/interface/web/search.html [hunk 2]
```diff
@@ -4,13 +4,13 @@
         <meta name="viewport" content="width=device-width, initial-scale=1.0 maximum-scale=1.0">
         <title>Khoj - Search</title>
 
-        <link rel="icon" type="image/png" sizes="128x128" href="/static/assets/icons/favicon-128x128.png">
-        <link rel="manifest" href="/static/khoj.webmanifest">
-        <link rel="stylesheet" href="/static/assets/khoj.css">
+        <link rel="icon" type="image/png" sizes="128x128" href="/static/assets/icons/favicon-128x128.png?v={{ khoj_version }}">
```
Comment (sabaimran) [inline src/khoj/interface/web/search.html:7]: Is the rationale behind including the `khoj_version` to limit the caching to the current version for static files? Neat solution.

**Review by sabaimran — APPROVED** [review]
Main question just about how to handle unknown file types, but all else looks good.

File: src/khoj/interface/web/search.html (hunk 2 above)
Comment (debanjum) [reply src/khoj/interface/web/search.html:7]: Yes, exactly!

File: src/interface/obsidian/src/utils.ts (hunk 1 above)
Comment (debanjum) [reply src/interface/obsidian/src/utils.ts]: Yeah, this is the same function as the one we use in desktop `main.js`. I've updated the code to `console.warn` when we fallback to the default option for now

**Pull request merged**